cc/keystone/auth/plugins/cc_x509.py

```diff
@@ -39,6 +42,37 @@
 LOG = log.getLogger(__name__)
 
 METHOD_NAME = 'external'
+URLENCODED_PEM_CERTIFICATE_PREFIX = b"-----BEGIN%20CERTIFICATE-----%0"
+
+
+
+def parse_certificate(pem_input: str | bytes) -> OpenSSL.crypto.X509:
+    """Parses formatted PEM input into OpenSSL.crypto.X509 certificate.
+
+    Supported PEM input formats:
+    * certificate as formatted by ingress-nginx, see https://github.com/kubernetes/ingress-nginx/blob/f820c51858ac98314ca535e091211ed13f1d5b62/docs/user-guide/nginx-configuration/annotations.md?plain=1#L284)
+    * certificate as formatted by Akamai, see "Cert PEM No Labels" @ https://techdocs.akamai.com/property-mgr/docs/set-var-extraction
+    """
+
+    if isinstance(pem_input, str):
+        pem_input = pem_input.encode()
+
+    if pem_input.startswith(URLENCODED_PEM_CERTIFICATE_PREFIX):
```

Comment:
Could it be that we get an empty line in the certificate in some case? If yes, this won't work.